applications/StructuralMechanicsApplication/tests/constitutive_law_test.py

```diff
@@ -93,33 +61,127 @@ def test_Uniaxial_HyperElastic_3D(self):
+        
+        print("The Material Response PK2")
+        cl.CalculateMaterialResponsePK2(cl_params)
+        print("Stress = ", cl_params.GetStressVector())
+        print("Strain = ", cl_params.GetStrainVector())
+        print("C      = ", cl_params.GetConstitutiveMatrix())
+
+        cl.FinalizeMaterialResponsePK2(cl_params)
+        cl.FinalizeSolutionStep(properties, geom, N, model_part.ProcessInfo)
 
-        #print("\n The Material Response Kirchhoff")
-        #cl.CalculateMaterialResponseKirchhoff(cl_params)
-        #print("Stress = ", cl_params.GetStressVector())
-        #print("Strain = ", cl_params.GetStrainVector())
-        #print("C      = ", cl_params.GetConstitutiveMatrix())
+        print("\n The Material Response Kirchhoff")
+        cl.CalculateMaterialResponseKirchhoff(cl_params)
+        print("Stress = ", cl_params.GetStressVector())
+        print("Strain = ", cl_params.GetStrainVector())
+        print("C      = ", cl_params.GetConstitutiveMatrix())
 
-        #cl.FinalizeMaterialResponseKirchhoff(cl_params)
-        #cl.FinalizeSolutionStep(properties, geom, N, model_part.ProcessInfo)
+        cl.FinalizeMaterialResponseKirchhoff(cl_params)
+        cl.FinalizeSolutionStep(properties, geom, N, model_part.ProcessInfo)
 
-        #print("\n The Material Response Cauchy")
-        #cl.CalculateMaterialResponseCauchy(cl_params)
-        #print("Stress = ", cl_params.GetStressVector())
-        #print("Strain = ", cl_params.GetStrainVector())
-        #print("C      = ", cl_params.GetConstitutiveMatrix())
+        print("\n The Material Response Cauchy")
+        cl.CalculateMaterialResponseCauchy(cl_params)
+        print("Stress = ", cl_params.GetStressVector())
+        print("Strain = ", cl_params.GetStrainVector())
+        print("C      = ", cl_params.GetConstitutiveMatrix())
         
-        #cl.FinalizeMaterialResponseCauchy(cl_params)
-        #cl.FinalizeSolutionStep(properties, geom, N, model_part.ProcessInfo)
+        cl.FinalizeMaterialResponseCauchy(cl_params)
+        cl.FinalizeSolutionStep(properties, geom, N, model_part.ProcessInfo)
+    
+    def test_Uniaxial_HyperElastic_3D(self):
+        nnodes = 4
+        dim = 3
+
+        # Define a model  and geometry
+        model_part = KratosMultiphysics.ModelPart("test")
+        geom = self._create_geometry(model_part, dim, nnodes)
+
+        # Material properties
+        young_modulus = 200e9
+        poisson_ratio = 0.3
+        properties = self._create_properties(model_part, young_modulus, poisson_ratio)
+
+        N = KratosMultiphysics.Vector(nnodes)
+        DN_DX = KratosMultiphysics.Matrix(nnodes, dim)
+
+        # Construct a constitutive law 
+        cl = StructuralMechanicsApplication.HyperElastic3DLaw()
+        self._cl_check(cl, properties, geom, model_part, dim)
+
+        # Set the parameters to be employed
+        dict_options = {'USE_ELEMENT_PROVIDED_STRAIN': False, 
```

Comment:
wouldn't it be better to set it directly here??

Reply:
I think is more modular to do it this way

Reply:
Honestly I disagree... @pooyan-dadvand what do you think?